test/web_socket_task_test.rb

```diff
@@ -0,0 +1,214 @@
+# frozen_string_literal: true
+
+require 'kontena-websocket-client'
+require 'json'
+
+using_task_library 'controldev'
+
+WebsocketStruct = Concurrent::MutableStruct.new :ws,
+                  :received_messages, :current_state, :has_error
+
+def websocket_connect(state)
+    Kontena::Websocket::Client.connect(@url) do |ws|
+        state.ws = ws
+        state.current_state = :open
+
+        ws.read do |message|
+            state.received_messages.append(message)
+        end
+    end
+    rescue Kontena::Websocket::CloseError => exc
+        state.current_state = :closed
+    rescue Kontena::Websocket::Error => exc
+        state.has_error = true
+end
+
+describe OroGen.controldev.WebSocketTask do
+    run_live
+
+    def start_websocket (name)
+        $first_websockets[name] = WebsocketStruct.new(nil, [], :pending, false)
+
+        thread = Thread.new {
```

Comment:
The convention is to use `{ }` for single-line blocks and `do ... end` for multiline blocks. You've done this a few times, I won't be repeating it.